internals/cli/app.go

```diff
@@ -185,15 +185,15 @@ func (a *App) PersistentFlags() *FlagSet {
 // registerRootEnvVarParsing ensures that flags on the root command with environment variables are set to
 // the value of their corresponding environment variable if they are not set already.
 func (a *App) registerRootEnvVarParsing() {
-	a.Root.Cmd.PersistentPreRunE = MergeFuncs(a.Root.Cmd.PersistentPreRunE, func(_ *cobra.Command, _ []string) error {
+	a.Root.Cmd.PersistentPreRunE = func(_ *cobra.Command, _ []string) error {
 		for _, flag := range a.Root.flags {
 			err := setFlagFromEnv(flag)
 			if err != nil {
 				return err
 			}
 		}
 		return nil
-	})
+	}
```

Comment:
One thing to note here, is that this code is now less resilient because it no longer gracefully handles the sitatuation where `PersistentPreRunE` is not `nil`. No direct need to change it, but good to be aware of this.

Reply:
Fixed by adding `AddPersistentPreRunE` as a new function.